css-scroll-snap-2/Overview.bs

```diff
@@ -196,90 +196,31 @@ The 'scroll-start-target' property {#scroll-start-target}
+		<a>apply initial scroll targets</a>.
 
 		The steps to <dfn>apply initial scroll targets</dfn> for a <a>scroll container</a>
 		|scrollcontainer| are as follows:
 
-		1. Let |blocktarget| be the <a>initial scroll target</a> for |scrollcontainer| in the block axis.
-		1. Let |inlinetarget| be the <a>initial scroll target</a> for |scrollcontainer| in the inline axis.
-		1. If |blocktarget| is not null:
-			1. Set the <a>block axis</a> component of |scrollcontainer|’s <a>initial scroll position</a>
-				to the result of running the steps to <a>get initial scroll offset</a> for
-				|scrollcontainer| with the axis parameter as “block” and the target parameter
-				as |blocktarget|.
-		1. If |inlinetarget| is not null:
-			1. Set the <a>inline axis</a> component of |scrollcontainer|’s <a>initial scroll position</a>
-				to the result of running the steps to <a>get initial scroll offset</a> for
-				|scrollcontainer| with the axis parameter as “inline” and the target parameter
-				as |inlinetarget|.
-
-		The steps to <dfn>get initial scroll offset</dfn> for a <a>scroll container</a> |scrollcontianer|,
-		given axis |axis| (which is either “block” or “inline”) and target |target| are as follows:
-
-		1. Let |alignment| be ''start''.
-		1. If |axis| is "block":
-			1. If the value of |target|'s [[css-scroll-snap-1#scroll-snap-align|scroll-snap-align]]
-				property along the block axis is not ''none'':
-				1. Let |alignment| be the value of |target|'s [[css-scroll-snap-1#scroll-snap-align|scroll-snap-align]]
-					property along the block axis.
-			1. Return the nearest <a>scroll offset</a> that aligns |scrollcontainer| with
-				|target| along the block axis according to |alignment|.
-		1. Otherwise:
-			1. If the value of |target|'s [[css-scroll-snap-1#scroll-snap-align|scroll-snap-align]]
-				property along the inline axis is not ''none'':
-					1. Let |alignment| be the value of |target|'s [[css-scroll-snap-1#scroll-snap-align|scroll-snap-align]]
-						property along the inline axis.
-			1. Return the nearest <a>scroll offset</a> that aligns |scrollcontainer| with
-				|target| along inline axis according to |alignment|.
-
-	<h4 id="scroll-start-target-longhands">Longhands</h4>
-
-		<pre class="propdef">
-			Name: scroll-start-target-block, scroll-start-target-inline, scroll-start-target-x, scroll-start-target-y
-			Value: auto | none
-			Initial: none
-			Applies to: all elements
-			Inherited: no
-			Logical property group: scroll-start-target
-			Percentages: n/a
-			Computed Value: either of the keywords "none" or "auto"
-			Animation type: not animatable
-		</pre>
+		1. Let |target| be the <a>initial scroll target</a> for |scrollcontainer|.
+		1. If |target| is null, return.
+		1. Run the steps to <a spec="cssom-view-1" lt="scroll a target into view">scroll |target| into view</a>
+			with |behavior| set to "auto", |block| set to "start", |inline| set to "nearest" and
```

Comment:
So I think we want to share like 95% of the algorithm, except for the last step where it says:

> 6. [Perform a scroll](https://drafts.csswg.org/cssom-view/#perform-a-scroll) of box to position, element as the associated element and behavior as the scroll behavior.

As we don't want this to be an actual scroll, but rather just the initial position of the scroller. Note it can actually perform a scroll if the scroll target turns up late in the loading process where that's unavoidable.

As such, we should pull out all of the steps which determine the target position into an algorithm we can invoke here to calculate the initial scroll position. I think this might also resolve the issue of needing to only scroll the nearest scroller without needing to add the flag.

Reply:
Sounds good, I've taken a stab at extracting the portion of the algorithm that determines the position into a set of steps to "determine the scroll-view position". Lmk wyt.